src/ui_dialogs.ts

```diff
@@ -117,17 +175,92 @@ export function openEditLinkDialog(column:LinkColumn, $header:d3.Selection<Colum
   });
 }
 
+
+// Renderer type change
+export function renderertypedialog(column: Column, $header: d3.Selection<Column>) {
+  var renderertype = column.rendererType();
+  const rendererTypelist = column.getRendererList();
+
+  var popup = makesortPopup($header, 'Change Visualization </br>', rendererTypelist.map(function (d, i) {
+    return `<input type="radio" name="renderertype" value=${d.type}  ${(renderertype === d.type) ? 'checked' : ''}> ${d.label}<br>`;
+
+  }).join('\n'));
+
+
+  function thiselement() {
+    return this === (<any>d3.event).target;
+
+  }
+
+  var that;
+  var renderercontent = d3.selectAll('input[name="renderertype"]');
+
+  renderercontent.on('change', function () {
+    that = this;
+    renderertype = that.value;
+    column.setRendererType(that.value);
+
+  });
+
+  d3.select('body').on('click', function () {
+    var outside = renderercontent.filter(thiselement).empty();
```

Comment:
it is not obvious how this function works, what does the filter `thiselement` a comment when the condition `outside` is true or false might help